### PR TITLE
Singleton Pattern

### PR DESCRIPTION
**Pull request by kanagalakshmi74**
Implemented Singleton pattern

### CODE REVIEW

File: LoginDemo/LearningSelenium/SingletonTest.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        public void TestLoginFail()
+        {
+            Singleton singletonObj = Singleton.GetInstance(driver);
+            driver.Navigate().GoToUrl("https://opensource-demo.orangehrmlive.com/web/index.php/auth/login");
+            utilityObj.SetImplicitWait(20);
+            singletonObj.EnterUserName("admin");
+            singletonObj.EnterPassword("admin1234");
+            singletonObj.LoginClick();
+            utilityObj.SetExplicitWait(singletonObj.GetErrorMessage(),20);
+            Assert.That(singletonObj.GetErrorMessage(),Is.EqualTo("Invalid credentials") , "username or password is incorrect");
```
Comment (rudzkokseniya) [inline LoginDemo/LearningSelenium/SingletonTest.cs:34]: Better to use Assert.AreEqual

File: LoginDemo/LearningSelenium/Singletons/Purchase.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        {
+            driver.FindElement(lName).SendKeys(lastName);
+        }
+
+        public void EnterPostal(string postalCode)
+        {
+            driver.FindElement(postal).SendKeys(postalCode);
+        }
+
+        public void ContinueButton()
```
Comment (rudzkokseniya) [inline LoginDemo/LearningSelenium/Singletons/Purchase.cs:43]: Method names should describe an action(Click, Press, Enter, etc)

File: LoginDemo/LearningSelenium/SingletonTest.cs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        {
+            TestLogin();
+            Product productObj = Product.GetInstance(driver);
+            productObj.GetItem1();
+            productObj.LinkItem1();
+            driver.FindElement(productObj.backMenu).Click();
+            productObj.GetItem2();
+            productObj.LinkItem2();
+            driver.FindElement(productObj.backMenu).Click();
+            utilityObj.SetExplicitWait(productObj.GetItemCount().ToString(), 20);
```
Comment (rudzkokseniya) [inline LoginDemo/LearningSelenium/SingletonTest.cs:75]: Next step:
Learn about  Factory Method pattern.
Read about test data and config data files: Test data - kind of data, that test is built one, like name of a game, country in registration, etc.

Config data - kind of data that is used to configure a project in whole (URL, browser name, wait time, etc.):

How to store data - Data should be stored separately (config and test) in xml or json file (chosen format should be used in a whole project).

Reading data - For reading config and test data separate utility class should be implemented;

Data - Any number different from 0 and 1 if it's used in a project should stored as a constant, if it's not suitable to be took out to test/config data

File: LoginDemo/LearningSelenium/SingletonTest.cs (hunk 1 above)
Comment (kanagalakshmi74) [reply LoginDemo/LearningSelenium/SingletonTest.cs:34]: I have used Assert.AreEqual but I got suggestion to change Asset.That from visual studio

File: LoginDemo/LearningSelenium/Singletons/Purchase.cs (hunk 2 above)
Comment (kanagalakshmi74) [reply LoginDemo/LearningSelenium/Singletons/Purchase.cs:43]: Ok Let me change it as ContinueClick()

File: LoginDemo/LearningSelenium/SingletonTest.cs (hunk 3 above)
Comment (kanagalakshmi74) [reply LoginDemo/LearningSelenium/SingletonTest.cs:75]: Factory Method, test data config data will be implemented in next task

**Pull request merged**